### PR TITLE
Type check added for FloatImm

### PR DESCRIPTION
**Pull request by ANSHUMAN87**

### CODE REVIEW

File: tests/python/unittest/test_target_custom_datatypes.py [hunk 1]
```diff
@@ -128,7 +128,7 @@ def test_bfloat_add_and_cast_FloatImm():
     Z = topi.cast(
         topi.add(
             topi.cast(X, dtype="custom[bfloat]16"),
-            tvm.tir.FloatImm("custom[bfloat]16", 1.5)),
+            tvm.tir.FloatImm("float", 1.5)),
```
Comment (tqchen) [inline tests/python/unittest/test_target_custom_datatypes.py:131]: There is a reason why we did not add type check to the codebas here(to support custom data types) as show in here.

**Comment by tqchen** [comment]
Thanks @ANSHUMAN87 .There is a reason why we did not add type check to the codebas here(to support custom data types) as show in here. 

Closing for now as the PR does not corresponds to a case that we would like to fix.

File: tests/python/unittest/test_target_custom_datatypes.py (hunk 1 above)
Comment (ANSHUMAN87) [reply tests/python/unittest/test_target_custom_datatypes.py:131]: I am sorry, i did not understand completely the logic behind it. I mean how FloatImm can support datatype other than float. It looks like some potential hole. That is why i added check. Anyway if it is on purpose, then i will ignore it.

**Comment by ANSHUMAN87** [comment]
> Thanks @ANSHUMAN87 .There is a reason why we did not add type check to the codebas here(to support custom data types) as show in here.
> 
> Closing for now as the PR does not corresponds to a case that we would like to fix.

I believe other error message correction can be taken, those were confusing when i was debugging one issue.

**Comment by tqchen** [comment]
The other message corrections looks good, please send another PR :)

**Pull request closed without merging**